lact-daemon/src/server/gpu_controller.rs

```diff
@@ -131,16 +132,16 @@ pub struct PciSlotInfo {
 }
 
 #[cfg(feature = "nvidia")]
-pub type NvidiaLibs = (Rc<Nvml>, Rc<Option<NvApi>>);
+pub type NvidiaLibs = (Arc<Nvml>, Arc<Option<NvApi>>);
 #[cfg(not(feature = "nvidia"))]
 pub type NvidiaLibs = ();
 
 pub(crate) fn init_controller(
     path: PathBuf,
     pci_db: &pciid_parser::Database,
-    nvml: &LazyCell<Option<NvidiaLibs>>,
-    amd_drm: &LazyCell<Option<LibDrmAmdgpu>>,
-    intel_drm: &LazyCell<Option<Rc<IntelDrm>>>,
+    nvml: &'static Option<NvidiaLibs>,
+    amd_drm: &'static Option<LibDrmAmdgpu>,
+    intel_drm: &'static Option<IntelDrm>,
```

Comment:
This is incorrect - each respective library should only attempt to be loaded when a controller actually needs it. Right now it will print an error in logs how NVML couldn't be loaded even on a system that doesn't have an nvidia card.

But in general, I don't think it makes sense to even pass the libs as static references, you can just mark  the LazyLocks `pub(crate)` and access them from here or from controllers directly (though it's a good idea to still store the reference in the controller to avoid Option checks on every access)